src/main/java/org/isf/malnutrition/rest/MalnutritionController.java

```diff
@@ -0,0 +1,136 @@
+		logger.info("Creating a new malnutrition ...");
+		boolean isCreated = manager.newMalnutrition(mapper.map2Model(malnutritionDTO));
+		if (!isCreated) {
+			logger.info("Malnutrition is not created!");
+            throw new OHAPIException(new OHExceptionMessage(null, "Malnutrition is not created!", OHSeverityLevel.ERROR));
+        }
+		logger.info("Malnutrition successfully created!");
+        return ResponseEntity.status(HttpStatus.CREATED).body(null);
+	}
+	
+	/**
+	 * Retrieves all the {@link Malnutrition} associated to the given admission id.
+	 * @param admissionID the admission id to use as filter.
+	 * @return all the retrieved malnutrition.
+	 * @throws OHServiceException 
+	 */
+	@GetMapping(value = "/malnutritions/{id_admission}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<MalnutritionDTO>> getMalnutrition(@PathVariable("id_admission") String admissionID) throws OHServiceException{
+		logger.info("Looking for malnutrition controls. Admission ID is " + admissionID);
+		List<Malnutrition> malnutritions = manager.getMalnutrition(admissionID);
+		if(malnutritions == null) {
+			throw new OHAPIException(new OHExceptionMessage(null, "Error while retrieving malnutrition controls!", OHSeverityLevel.ERROR));
+		}
+		List<MalnutritionDTO> mappedMalnutritions = mapper.map2DTOList(malnutritions);
+		if(mappedMalnutritions.isEmpty()) {
+			logger.info("No malnutrition control found");
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(mappedMalnutritions);
+		} else {
+			logger.info("Found " + mappedMalnutritions.size() + " malnutrition controls");
+			return ResponseEntity.ok(mappedMalnutritions);
+		}
+	}
+	
+	/**
+	 * returns the last {@link Malnutrition} entry for specified patient ID
+	 * @param patientID - the patient ID
+	 * @return the last {@link Malnutrition} for specified patient ID.
+	 * @throws OHServiceException 
+	 */
+	@GetMapping(value = "/malnutritions/last/{id_patient}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<MalnutritionDTO> getLastMalnutrition(@PathVariable("id_patient") int patientID) throws OHServiceException {
+		Malnutrition foundMalnutrition = manager.getLastMalnutrition(patientID);
+		if(foundMalnutrition == null) {
+            throw new OHAPIException(new OHExceptionMessage(null, "No malnutrition found!", OHSeverityLevel.ERROR));
+		} else {
+			return ResponseEntity.ok(mapper.map2DTO(foundMalnutrition));
+		}
+	}
+	
+	/**
+	 * Updates the specified malnutrition
+	 * @param the {@link Malnutrition} to update
+	 * @return the updated {@link Malnutrition}
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/malnutritions", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<MalnutritionDTO> updateMalnutrition(@RequestBody @Valid MalnutritionDTO malnutritionDTO) throws OHServiceException {
+		Malnutrition updatedMalnutrition = manager.updateMalnutrition(mapper.map2Model(malnutritionDTO));
```

Comment:
Would it make sense to add an existence check before calling update?
I can't see any existance check in the method implementation.